actioncable/lib/action_cable/subscription_adapter/postgresql.rb

```diff
@@ -3,20 +3,24 @@
 gem "pg", "~> 1.1"
 require "pg"
 require "openssl"
+require "zlib"
 
 module ActionCable
   module SubscriptionAdapter
     class PostgreSQL < Base # :nodoc:
       prepend ChannelPrefix
+      DEFAULT_COMPRESS_LIMIT = 1.kilobyte
 
       def initialize(*)
         super
         @listener = nil
+        @compress_payloads = @server.config.cable[:compress_payloads]
       end
 
       def broadcast(channel, payload)
+        payload = Zlib.deflate(payload) if @compress_payloads && payload.size > DEFAULT_COMPRESS_LIMIT
```

Comment:
Let's use `ActiveSupport::Gzip` here, it would simplify the code a bit: https://github.com/rails/rails/blob/main/activesupport/lib/active_support/gzip.rb